source/fundamentals/connection/mongoclientsettings.txt

```diff
@@ -0,0 +1,197 @@
+====================
+MongoClient Settings
+====================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+
+Overview
+--------
+
+In this guide, you can learn how to control your connections' behavior
+in a ``MongoClient``.
+
+You can control your connections' behavior by creating and passing in a 
+`MongoClientSettings <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.html>`__
+object to the `MongoClients.create() <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/MongoClients.html#create(com.mongodb.MongoClientSettings)>`__
+method.
+
+To create a ``MongoClientSettings`` object, use the
+``MongoClientSettings.builder()`` method. The builder makes the
+settings immutable and supports a simpler construction through chaining.
+After chaining all the settings you want, use the ``build()`` method to
+create the ``MongoClientSettings`` object.
+
+Example
+~~~~~~~
+
+This example demonstrates specifying a ``ConnectionString`` in a
+``MongoClientSettings``:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/mcs.java
+   :start-after: begin ConnectionString
+   :end-before: end ConnectionString
+   :language: java
+   :emphasize-lines: 3
+   :dedent:
+
+.. tip::
+
+   If you omit the ``applyConnectionString()`` method, you connect to
+   your localhost.
+
+For a list of methods available in ``MongoClientSettings``, 
+see the `API Documentation <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.Builder.html>`__.
+
+The following sections describe the commonly used ``MongoClientSettings``:
+
+- :ref:`Cluster Settings <mcs-cluster-settings>`
+- :ref:`Socket Settings <mcs-socket-settings>`
+- :ref:`Connection Pool Settings <mcs-connectionpool-settings>`
+- :ref:`Server Settings <mcs-server-settings>`
+- :ref:`SSL Settings <mcs-ssl-settings>`
+
+.. _mcs-cluster-settings:
+
+Cluster Settings
+----------------
+
+Use the `applyToClusterSettings() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.Builder.html#applyToClusterSettings(com.mongodb.Block)>`__
+method to specify settings on how to connect to a MongoDB cluster. 
+
+Example
+~~~~~~~
+
+This example specifies the following settings for connecting to a
+MongoDB cluster:
+
+- To use the settings from your ``ConnectionString``
+- To connect to ``MULTIPLE`` servers in your cluster
```

Comment:
question: Why would this be necessary? Is it required if I'm specifying a replicaset or srv uri in the connection string?